parser.cpp

```diff
@@ -634,53 +664,66 @@ Attrs* Parser::parse_attributes(bool fcn)
                              false,
                              Token::empty(),
                              Token::empty()};
+    attr_toks_.clear();
     do
     {
         Token t = next();
-        if (fcn && t == "in" && !(attrs->in_ || attrs->inout_))
-            *(attrs->out_ ? &attrs->inout_ : &attrs->in_) = true;
-        else if (fcn && t == "out" && !(attrs->out_ || attrs->inout_))
-            *(attrs->in_ ? &attrs->inout_ : &attrs->out_) = true;
-        else if (fcn && t == "string" && _no_kind(attrs))
-            attrs->string_ = true;
-        else if (fcn && t == "wstring" && _no_kind(attrs))
-            attrs->wstring_ = true;
-        else if (fcn && t == "isptr" && _no_kind(attrs))
-            attrs->isptr_ = true;
-        else if (fcn && t == "isary" && _no_kind(attrs))
-            attrs->isary_ = true;
-        else if (t == "user_check" && !attrs->user_check_)
-            attrs->user_check_ = true;
-        else if (t == "count" || t == "size")
+        AttrTok atok = check_attribute(t);
+
+        // Only count and size attributes are valid for struct properties.
+        if (!fcn && (atok != TokCount && atok != TokSize))
```

Comment:
Does this mean we still allow size and count for `enum`?

Reply:
Should we also support `string`?

Reply:
> Does this mean we still allow size and count for `enum`?

For enums it is a syntax error to specify attributes. Not supported by the grammar.

For unions, we check and error out if attributes have been specified.
Added tests.

Reply:
> Should we also support `string`?

That is a feature that we can add in the future. Maybe before that we could switch to a meta data based marshaling approach.